delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchIndexingCoordinator.scala

```diff
@@ -0,0 +1,157 @@
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.sdk.eventlog.GlobalEventLog
+import ch.epfl.bluebrain.nexus.delta.sdk.indexing.IndexingStreamCoordinator
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment.IriSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, ResourceF}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.sourcing.projections.ProjectionId.ViewProjectionId
+import ch.epfl.bluebrain.nexus.sourcing.projections.ProjectionStream.{ChunkStreamOps, SimpleStreamOps}
+import ch.epfl.bluebrain.nexus.sourcing.projections.stream.StreamSupervisor
+import ch.epfl.bluebrain.nexus.sourcing.projections.{Message, Projection, ProjectionProgress}
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import io.circe.Json
+import monix.bio.{IO, Task}
+import monix.execution.Scheduler
+import org.apache.jena.rdf.model.Property
+
+private class IndexingStream(
+    client: ElasticSearchClient,
+    viewRes: IndexingViewResource,
+    config: ElasticSearchViewsConfig
+)(implicit cr: RemoteContextResolution, baseUri: BaseUri) {
+  private val view: IndexingElasticSearchView         = viewRes.value
+  private val index: IndexLabel                       = IndexLabel.fromView(config.indexing.prefix, view.uuid, viewRes.rev)
+  private val ctx: ContextValue                       = ContextValue(contexts.elasticSearchIndexing)
+  private val originalSource: Property                = predicate(nxv.originalSource.iri)
+  implicit private val projectionId: ViewProjectionId = viewRes.projectionId
+
+  private def deleteOrIndex(res: ResourceF[IndexingData]): Task[ElasticSearchBulk] =
+    if (res.deprecated && !view.includeDeprecated) delete(res)
+    else index(res)
+
+  private def delete(res: ResourceF[IndexingData]): Task[ElasticSearchBulk] =
+    Task.pure(ElasticSearchBulk.Delete(index, res.id.toString))
+
+  private def index(res: ResourceF[IndexingData]): Task[ElasticSearchBulk] =
+    toDocument(res).map(doc => ElasticSearchBulk.Index(index, res.id.toString, doc))
+
+  private def toDocument(res: ResourceF[IndexingData]): Task[Json] = {
+    val g = res.value.selectPredicatesGraph
+    (if (view.includeMetadata) res.void.toGraph.mapError(illegalArgument).map(_ ++ g) else Task.pure(g)).flatMap {
+      case graph if view.sourceAsText =>
+        val jsonLd = graph.add(originalSource, obj(res.value.source.noSpaces)).toCompactedJsonLd(ctx)
+        jsonLd.mapError(illegalArgument).map(_.json.removeKeys(keywords.context))
+      case graph                      =>
+        val jsonLd = graph.toCompactedJsonLd(ctx)
+        jsonLd.mapError(illegalArgument).map(ld => res.value.source deepMerge ld.json.removeKeys(keywords.context))
+    }
+  }
+
+  private def illegalArgument[A](error: A) =
```

Comment:
Same as with Blazegraph, if we can have more meaningful errors

Reply:
Out of scope and not necessary.